app/models/ems_cluster.rb

```diff
@@ -182,6 +182,17 @@ def parent_datacenter
     detect_ancestor(:of_type => 'EmsFolder') { |a| a.kind_of?(Datacenter) }
   end
 
+  has_many :ems_events,
+            ->(cluster) {
+              where("ems_cluster_id" => cluster.id)
```

Comment:
at least in terms of how `ems_cluster_id` is concerned, can we use `foreign_key`?

And getting rid of the `or` seems to change this quite a bit

Reply:
Absolutely.  Great idea.  It feels like this was a POC for a fix for the first issue of not being able to show cluster subclasses on the timelines.  I had a test for the base cluster class but subclasses weren't tested so I recreated it with them and was able to fix it with the code above.

For the second question, do we show other objects on the cluster timeline such as events from vms or hosts in the cluster, I think that's secondary question.  We would need to ensure the UI passes limits to display it properly.  I don't know if we should cowardly refuse to try to draw a timeline with 10,000+ events on it.  For now, I was going to try to fix the bug first and then open it up for discussion on how to properly display it for large number of objects/events.